src/test/java/org/minions/devfund/marylin/Squarelotron.java

```diff
@@ -0,0 +1,267 @@
+package org.minions.devfund.marylin;
+
+import java.util.Arrays;
+
+/**
+ * Squarelotron class.
+ */
+public class Squarelotron {
+    private int[][] squarelotron;
+    private int size;
+    private int numberOfRings;
+
+    /**
+     * Constructor of the matrix n x n initialized with consecutive values from 1 to m.
+     *
+     * @param n size of the matrix
+     */
+    public Squarelotron(int n) {
+        this.size = n;
+        this.numberOfRings = this.size - (this.size / 2);
+        this.squarelotron = new int[n][n];
+        int number = 1;
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < n; j++) {
+                this.squarelotron[i][j] = number;
+                number++;
+            }
+        }
+    }
+
+    /**
+     * Constructs an Squarelotron from an array of array.
+     *
+     * @param squarelotron instance of an Squarelotron.
+     */
+    public Squarelotron(final int[][] squarelotron) {
+        this.size = squarelotron.length;
+        this.numberOfRings = this.size - (this.size / 2);
+        this.squarelotron = new int[this.size][this.size];
+        for (int i = 0; i < this.size; i++) {
+            for (int j = 0; j < this.size; j++) {
+                this.squarelotron[i][j] = squarelotron[i][j];
+                this.squarelotron[i][j] = squarelotron[i][j];
+            }
+        }
+    }
+
+
+    @Override
+    public int hashCode() {
+        return Arrays.hashCode(squarelotron);
+    }
+
+    /**
+     * Compare two Squarelotron.
+     *
+     * @param object Squarelotron instance.
+     * @return Squarelotron instance.
+     */
+    @Override
+    public boolean equals(final Object object) {
+        if (!(object instanceof Squarelotron)) {
+            return false;
+        }
+
+        Squarelotron squarelotronTemp = (Squarelotron) object;
+        for (int i = 0; i < this.squarelotron.length; i++) {
+            for (int j = 0; j < this.squarelotron.length; j++) {
+                if (this.squarelotron[i][j] != squarelotronTemp.squarelotron[i][j]) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Duplicates an array of arrays.
+     *
+     * @param original array of array.
+     * @return the duplicated array.
+     */
+    private int[][] duplicateGrid(final int[][] original) {
+        int size = original.length;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "size" which hides the field declared at line 10. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)